### DISCUSSION TITLE
Bug? Watch Field w/ shouldUnregister false and w/ FormProvidder

### DISCUSSION THREAD

**Discussion by fadi-george** [react-hook-form/react-hook-form#2982] [Q&A] [answered]
Hello, how would I watch the value of a field array in the parent component when field array lives in a child component?

With v.6.8.4, I'm getting undefined when the field array gets updated.
Click insert at or append then duplicate.
https://codesandbox.io/s/react-hook-form-usefieldarray-forked-34kqg?file=/src/index.js

**Comment by fadi-george** [comment, marked as answer]
With shouldUnregister, it'll just return the default value ([]).

**Reply by bluebill1049** [reply]
hey @fadi-george i think there is a misunderstanding with watch/useWatch, you will always need to have a defaultValue when you using useFieldArray methods. It was made to subscribe to the input changes, not the same as `state`

**Reply by bluebill1049** [reply]
<img width="984" alt="Screen Shot 2020-09-23 at 2 00 05 pm" src="https://user-images.githubusercontent.com/10513364/93965284-17dcd100-fda5-11ea-9e7a-fee702c29cf3.png">

I think i probably need to do a section with watch/useWatch vs state

**Reply by fadi-george** [reply]
I see, my bad. Seems I would have to pass down the fields to the child component and watch from parent.

**Reply by bluebill1049** [reply]
not your bad at all :) i think there is something i can do in the doc to make this clearer, feel free to give suggestions as well.
